addons/sale/models/sale.py

```diff
@@ -996,6 +996,13 @@ def _amount_by_group(self):
                             res[group]['amount'] += t['amount']
                             res[group]['base'] += t['base']
             res = sorted(res.items(), key=lambda l: l[0].sequence)
+
+            # round amount and prevent -0.00
+            rnd = currency.round
+            for l in res:
+                l[1]['amount'] = rnd(l[1]['amount']) + 0.0
+                l[1]['base'] = rnd(l[1]['base']) + 0.0
```

Comment:
```suggestion
            for group_data in res:
                group_data[1]['amount'] = currency.round(l[1]['amount']) + 0.0
                group_data[1]['base'] = currency.round(l[1]['base']) + 0.0
```
also you probably need to update the same code in account:
https://github.com/odoo/odoo/blob/bdc44341326e284cb27641afc9306e60c5b153d9/addons/account/models/account_move.py#L1704

Reply:
@smetl updated, don't need for account.move